addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2824,6 +2824,191 @@ options.registry.ReplaceImage = options.Class.extend({
+            }
+        }
+        // Creates CSS selectors based on those attributes, the reducers
+        // combine the attributes' values.
+        // Important to note how the CSS Rule works : we hide the content
+        // that match the selector, even for "Visible to" option.
+        // Therefore we hide when the HTML tag does not contain the values
+        // we're looking for
+        let selectors = '';
+        for (const attribute of onlyAttributes) {
+            // e.g of selector :
+            // htm:not([data-attr-1="valueAttr1"]) \
+            // :not([data-attr-1="valueAttr2"]) [data-visibility-id="ruleId"]
+            const selector = attribute.records.reduce((acc, record) => {
+                return acc += `:not([${attribute.name}="${record.value}"])`;
+            }, 'html') + ` body:not(.editor_enable) [data-visibility-id="${visibilityId}"]`;
+            selectors += selector + ', ';
+        }
+        for (const attribute of hideAttributes) {
+            // html[data-attr-1="valueAttr1"] [data-visibility-id="ruleId"],
+            // html[data-attr-1="valueAttr2"] [data-visibility-id="ruleId"]
+            const selector = attribute.records.reduce((acc, record, i, a) => {
+                acc += `html[${attribute.name}="${record.value}"] body:not(.editor_enable) [data-visibility-id="${visibilityId}"]`;
+                return acc + (i !== a.length - 1 ? ',' : '');
+            }, '');
+            selectors += selector + ', ';
+        }
+        selectors = selectors.slice(0, -2);
+        if (selectors) {
+            this.$target[0].dataset.visibilitySelectors = selectors;
+        }
+
+        this.$target[0].dataset['visibilityId'] = visibilityId;
+        return this._super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Inserts or deletes record's id and value in target's data-attributes
+     * if no ids are selected, deletes the attribute
+     *
+     * @see this.selectClass for parameters
+     */
+    selectRecord(previewMode, widgetValue, params) {
+        const recordsData = JSON.parse(widgetValue);
+        if (!recordsData.length) {
+            delete this.$target[0].dataset[params.attributeName];
+            return;
+        }
+        this.$target[0].dataset[params.attributeName] = widgetValue;
+    },
+    /**
+     * Opens the toggler when 'conditional' is selected
+     *
+     * @override
+     */
+    selectDataAttribute(previewMode, widgetValue, params) {
+        this._super(...arguments);
+        if (params.attributeName === 'visibility') {
+            if (widgetValue === 'conditional') {
+                const collapseEl = this.$el.children('we-collapse')[0];
+                this._toggleCollapseEl(collapseEl);
+                this.trigger_up('snippet_option_visibility_update', { show: true });
```

Comment:
I am missing something in this code... is not this event supposed to work only for `data-invisible="1"` fields? I can't remember. But this may have something to do with your need to remove the d-none class in `cleanForSave` as I explained in my other comment: I have the feeling your snippet is always `data-invisible="1"` while it should only be when conditional visibility.